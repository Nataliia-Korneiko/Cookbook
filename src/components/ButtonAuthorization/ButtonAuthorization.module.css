.button {
  width: 120px;
  height: 44px;
  border: 2px solid var(--color-golden);
  border-radius: 5px;
  font-family: 'Montserrat', sans-serif;
  font-size: 18px;
  font-weight: 500;
  color: var(--color-dark-blue);
  background-color: var(--bg-color-white);
  text-align: center;
  text-transform: uppercase;
  outline: none;
  cursor: pointer;
  transition: 0.8s ease-in-out;
  box-shadow: var(--box-shadow-light);
}

.button:hover,
.button:focus {
  background-color: var(--bg-color-golden);
  transition: 0.8s ease-in-out;
}
