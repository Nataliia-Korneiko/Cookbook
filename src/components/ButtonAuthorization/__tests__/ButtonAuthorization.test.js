/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { shallow } from 'enzyme';
import ButtonAuthorization from '../ButtonAuthorization';

describe('ButtonAuthorization component', () => {
  const props = {
    type: '',
    label: '',
    onClick: () => null,
  };

  const newButtonAdd = shallow(<ButtonAuthorization {...props} />);

  it('renders a properly', () => {
    expect(newButtonAdd).toMatchSnapshot();
  });

  it('renders one button', () => {
    expect(newButtonAdd.find('button')).toHaveLength(1);
  });

  describe('callback', () => {
    it('onClick prop has been called on click', () => {
      const mockCallBack = jest.fn();
      const wrapped = shallow(<ButtonAuthorization onClick={mockCallBack} />);

      wrapped.find('button').simulate('click');
      wrapped.update();
      expect(mockCallBack.mock.calls.length).toEqual(1);
    });
  });
});
