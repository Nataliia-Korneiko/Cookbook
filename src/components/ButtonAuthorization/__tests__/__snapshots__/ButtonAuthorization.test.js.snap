// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonAuthorization component renders a properly 1`] = `ShallowWrapper {}`;
